Automattic/wp-calypso
# Plugins Category Results routes now renders the PluginsCategoryResults page.

#### Proposed Changes

* Extracts `PluginsCategoryResults` page from `PluginsBrowser`.
* Categories route now render the `PluginsSearchResults` page.
* Redirects unsupported categories urls to the discovery page.
* Fixes an issue where the search wasn't cleared from the search input when hitting the clear button.

#### Testing Instructions



* Smoke test categories and the results page
* Smoke test the rest of the views (Discovery Page, Search Results Page).
* It should return the correct results
* It should send the correct tracks events
* visiting a url with an unsupported category like `hello` should redirect to the discovery page.

#### Pre-merge Checklist






Fixes #64547
